### DISCUSSION TITLE
Add Support for Delete Tracking in Incremental Append Mode

### DISCUSSION THREAD

**Discussion by nimuriel** [airbytehq/airbyte#45898] [Connector Ideas and Features]
Currently, when using incremental append mode with a cursor field (e.g., updated_at), deleted records are not tracked, leading to a loss of visibility on data deletions. I propose introducing a feature similar to Fivetran's "_fivetran_deleted" mechanism, where deleted records are flagged without requiring changes to the updated_at field. This feature would provide an option to track hard deletes by marking records, using the primary key, after deletion.

Existing alternatives, such as modifying the database to track soft deletes or using CDC (Change Data Capture), are either not feasible for large datasets or lead to an overflow of unnecessary records in the data warehouse, especially for high-change-rate tables. A simple delete-tracking mechanism in incremental append mode would address these issues without the need for full CDC or extensive database modifications.

**Comment by nathanfreystaetter** [comment]
Agree with this. Big need. Surprised this is not standard.

[a comment by an automated account is not shown]
